src/node_contextify.cc

```diff
@@ -445,268 +447,277 @@ class ContextifyContext {
-  static void New(const FunctionCallbackInfo<Value>& args) {
-    Environment* env = Environment::GetCurrent(args);
+void ContextifyScript::New(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
 
-    if (!args.IsConstructCall()) {
-      return env->ThrowError("Must call vm.Script as a constructor.");
-    }
-
-    ContextifyScript* contextify_script =
-        new ContextifyScript(env, args.This());
+  if (!args.IsConstructCall()) {
+    return env->ThrowError("Must call vm.Script as a constructor.");
+  }
 
-    TryCatch try_catch;
-    Local<String> code = args[0]->ToString(env->isolate());
-    Local<String> filename = GetFilenameArg(args, 1);
-    bool display_errors = GetDisplayErrorsArg(args, 1);
-    if (try_catch.HasCaught()) {
-      try_catch.ReThrow();
-      return;
-    }
+  ContextifyScript* contextify_script =
+      new ContextifyScript(env, args.This());
+  contextify_script->persistent().SetWrapperClassId(
+      ClassId::CONTEXTIFY_SCRIPT);
+
+  TryCatch try_catch;
+  Local<String> code = args[0]->ToString(env->isolate());
+  Local<String> filename = GetFilenameArg(args, 1);
+  bool display_errors = GetDisplayErrorsArg(args, 1);
+  if (try_catch.HasCaught()) {
+    try_catch.ReThrow();
+    return;
+  }
 
-    ScriptOrigin origin(filename);
-    ScriptCompiler::Source source(code, origin);
-    Local<UnboundScript> v8_script =
-        ScriptCompiler::CompileUnbound(env->isolate(), &source);
+  ScriptOrigin origin(filename);
+  ScriptCompiler::Source source(code, origin);
+  Local<UnboundScript> v8_script =
+      ScriptCompiler::CompileUnbound(env->isolate(), &source);
 
-    if (v8_script.IsEmpty()) {
-      if (display_errors) {
-        AppendExceptionLine(env, try_catch.Exception(), try_catch.Message());
-      }
-      try_catch.ReThrow();
-      return;
+  if (v8_script.IsEmpty()) {
+    if (display_errors) {
+      AppendExceptionLine(env, try_catch.Exception(), try_catch.Message());
     }
-    contextify_script->script_.Reset(env->isolate(), v8_script);
+    try_catch.ReThrow();
+    return;
   }
+  contextify_script->script_.Reset(env->isolate(), v8_script);
+}
 
 
-  static bool InstanceOf(Environment* env, const Local<Value>& value) {
-    return !value.IsEmpty() &&
-           env->script_context_constructor_template()->HasInstance(value);
-  }
+bool ContextifyScript::InstanceOf(Environment* env, const Local<Value>& value) {
+  return !value.IsEmpty() &&
+         env->script_context_constructor_template()->HasInstance(value);
+}
 
 
   // args: [options]
-  static void RunInThisContext(const FunctionCallbackInfo<Value>& args) {
-    // Assemble arguments
+void
+ContextifyScript::RunInThisContext(const FunctionCallbackInfo<Value>& args) {
+  // Assemble arguments
+  TryCatch try_catch;
+  uint64_t timeout = GetTimeoutArg(args, 0);
+  bool display_errors = GetDisplayErrorsArg(args, 0);
+  if (try_catch.HasCaught()) {
+    try_catch.ReThrow();
+    return;
```

Comment:
There are no actual changes to this file, see the OP:

> node-contextify.cc
> Extracted out ContextifyScript class declaration to a header. The diff looks horrible but there is no real changes
> Added check to prevent cancellation of termination in EvalMachine method